### ISSUE TITLE
Problem PATCHing a comment after it is created it. 

### ISSUE THREAD

**Issue by kpie89** [ga-wdi-boston/capstone-project#147]
When a user wants to add a song to a playlist, they comment on the playlist with their new song and a description. A user clicks the song they want which creates it and the rails server logs the POST in the rails and the necessary params show up including the playlist_id but when I try to edit the post the PUT returns a 404 and the playlist_id param is missing. 

I am passing the data up from a "new-comment-snippet" which draws upon the playlist model hook and transfers data all the way up to the new-comment route. But I suppose I am not passing it the right data. 

New comment snippet

```
<div id="playlist-list">
<hr>
        <label for="desc">Description</label>
        {{input type="desc"
                id="desc"
                placeholder="Description"
                value=comment.desc}}

<h2>{{playlist.song.title}}</h2>
<button class="btn btn-default" type="submit" {{action "updateComment" comment=comment}}>Make it</button>
</div>
```

new-comment-snippet component

```
import Ember from 'ember';

export default Ember.Component.extend({
  actions: {
    updateComment () {
      this.sendAction('updateComment', this.get('comment'));
    }
  }

  });
```

New comment template 

```
{{new-comment-snippet playlist=model updateComment="updateComment" comment=playlist.comment}}
{{outlet}}
```

And the new comment route

```
import Ember from 'ember';

export default Ember.Route.extend({
  model (params) {
    return this.store.findRecord('playlist', params.playlist_id);
  },


  isEditing: false,


  actions: {
    updateComment(comment) {
      console.log(comment);
      comment.save()
      .then(() => this.transitionTo('playlists'));
    }
   }
  });
```

**Comment by faetea** [comment]
In the new comment route are you trying to retrieve a single record or multiple records?

```
  model (params) {
    return this.store.findRecord('playlist', params.playlist_id);
  },
```

Use store.findRecord() to retrieve a SINGLE record by its type and ID.

**Comment by kpie89** [comment]
This problem was "fixed" by adding proper relationships in my back end. But @jrhorn424 I am again getting the problem 

```
"Assertion Failed: Passing classes to store methods has been removed. Please pass a dasherized string instead of undefined"
```

I have tried everything mentioned in the issues prior to this that have dealt with the same thing. I am very lost. I have done research on this issue and it seems to be new. But I have tried all the solutions laid out and nothing seems to be working.
